### DISCUSSION TITLE
Skip the retry functionality for specific tests or specFiles

### DISCUSSION THREAD

**Discussion by jurjandrei** [webdriverio/webdriverio#10679] [Q&A] [answered]
Hello,

I am using webdriverio and mocha and aiming to skip the retry mechanism for certain tests or spec files(usually this are tests which take long and don't need to be retried)

What have I tried until now:
1.   specFileRetries: 1,
  specFileRetryAttempts: 0, and incrementing it in the afterTest hook, from the same config file 
2. Using this.retries(number) at describe level block and .retries(number) at test level.
The afterTest hook looks something like
  afterTest(test, context, { error, passed, retries }) {
<code>
}

Is there a possibility to have the specFileRetries overwritten or not taken in consideration for specific spec files or tests? If yes, can someone please point me on what I am doing wrong above?

Thanks

**Comment by christian-bromann** [comment, marked as answer]
Hey @jurjandrei 👋 

this is currently not possible. How would you define which spec files should be retried or not. How would you overwrite the defined behavior?

**Reply by jurjandrei** [reply]
Thanks @christian-bromann for the info.
I would take use of test.parent and have something like [].includes(test.parent) to take the describe name and skip the retry with the fail message. Or instead of includes, to be able to set to particular tests or spec file the limit for retries.
Regarding the second question, is it possible please to mention what do mean by it? 
Thanks again

**Reply by christian-bromann** [reply]
> I would take use of test.parent and have something like [].includes(test.parent) to take the describe name and skip the retry with the fail message.

You will have file an issue in the Mocha or Jasmine project depending on what you are using.

> Regarding the second question, is it possible please to mention what do mean by it?

You were mentioning `specFileRetries` which is a WebdriverIO feature and I wondered if you have it set in the `wdio.conf.js` how would you want to overwrite it per particular spec file?

**Reply by jurjandrei** [reply]
I was thinking at first that maybe since having access in afterTest to retries, that specFileRetries will automatically populate the limit from the object and maybe I could set the attempts number inside the tests or in a before hook, throwing maybe the fail message since then the two values are equal. To be honest I didn't think about how would I implement it since I was thinking that maybe is currently possible to do so.
